orb_slam_2/src/ORBextractor.cc

```diff
@@ -1031,6 +1029,41 @@ void ORBextractor::ComputeKeyPointsOld(std::vector<std::vector<KeyPoint> > &allK
         computeOrientation(mvImagePyramid[level], allKeypoints[level], umax);
 }
 
+static bool isBorderOrInvalidCoordinate(const KeyPoint& keyPoint, const Mat& image)
+{
+  if (isnan(keyPoint.pt.x))
+    return true;
+  if (isnan(keyPoint.pt.y))
+    return true;
+
+  constexpr int minBorderX = EDGE_THRESHOLD - 3;
+  constexpr int minBorderY = EDGE_THRESHOLD - 3;
+  const int maxBorderX = image.cols - EDGE_THRESHOLD + 3;
+  const int maxBorderY = image.rows - EDGE_THRESHOLD + 3;
+
+  if (keyPoint.pt.x < minBorderX)
+    return true;
+  if (keyPoint.pt.x > maxBorderX)
+    return true;
+  if (keyPoint.pt.y < minBorderY)
+      return true;
+  if (keyPoint.pt.y > maxBorderY)
+    return true;
+
+
+  return false;
+}
+
+static void setRandomDescriptor(uchar* desc)
+{
+  for (int i = 0; i < 32; ++i)
+  {
+    int val;
+    val = std::rand() % 256;
+    desc[i] = (uchar)val;
```

Comment:
nit: this could be easily written in one line. Also, prefer static_cast<uchar>() according to conventions: https://google.github.io/styleguide/cppguide.html#Casting